extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/StreamReaderOptions.java

```diff
@@ -0,0 +1,25 @@
+package io.deephaven.extensions.barrage.util;
+
+import io.deephaven.annotations.SimpleStyle;
+import io.deephaven.extensions.barrage.BarrageSnapshotOptions;
+import io.deephaven.extensions.barrage.BarrageSubscriptionOptions;
+import io.deephaven.extensions.barrage.ColumnConversionMode;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@SimpleStyle
+public abstract class StreamReaderOptions {
+    public static StreamReaderOptions of(BarrageSnapshotOptions options) {
```

Comment:
Let's make this interface just an interface, and have the Barrage*Options classes implement it?